src/main/java/org/killbill/billing/plugin/vertex/VertexTaxCalculator.java

```diff
@@ -325,7 +325,9 @@ private InvoiceItem createTaxInvoiceItem(final InvoiceItem taxableItem, final UU
             taxRate = calculatedTax.divide(taxableItem.getAmount(), 5, RoundingMode.FLOOR).doubleValue();
         }
 
-        final String taxItemDetails = addTaxRateToItemDetails(taxItem.getItemDetails(), taxRate);
+        String parentItemID = adjustmentItem == null ? taxableItem.getId().toString() : adjustmentItem.getId().toString();
```

Comment:
parentItemID and additionalDetails can be marked as final

Reply:
done